src/rust/engine/src/nodes.rs

```diff
@@ -1292,6 +1286,15 @@ impl Node for NodeKey {
       _ => true,
     }
   }
+
+  fn user_facing_name(&self) -> Option<String> {
+    match self {
+      NodeKey::Task(ref task) => task.get_display_info().map(|s| s.to_owned()),
+      NodeKey::Snapshot(_) => Some(format!("{}", self)),
+      NodeKey::MultiPlatformExecuteProcess(mp_epr) => mp_epr.0.user_facing_name(),
+      _ => None,
```

Comment:
I'd advocate for explicitly enumerating the others here, so that if someone adds a new `NodeKey`, they're forced by the compile error to consider whether it should have a user-facing name.